plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/schema/confluent/ConfluentModule.java

```diff
@@ -146,13 +149,13 @@ public void configure(Binder binder)
         public void configure(Binder binder)
         {
             MapBinder<String, RowEncoderFactory> encoderFactoriesByName = encoderFactory(binder);
-            encoderFactoriesByName.addBinding(AvroRowEncoder.NAME).toInstance((session, dataSchema, columnHandles) -> {
-                throw new TrinoException(NOT_SUPPORTED, "Insert not supported");
-            });
-            encoderFactoriesByName.addBinding(ProtobufRowEncoder.NAME).toInstance((session, dataSchema, columnHandles) -> {
+            encoderFactoriesByName.addBinding(AvroRowEncoder.NAME).to(ConfluentRowEncoderFactory.class).in(SINGLETON);
+            encoderFactoriesByName.addBinding(KafkaRowDecoderFactory.NAME).to(KafkaSerializerRowEncoder.Factory.class).in(SINGLETON);
```

Comment:
I am really new to this topic, just started review, but cannot get why we bind decoder factory to encoder factory.

Reply:
Good catch, thanks!